include/nuttx/mqueue.h

```diff
@@ -85,25 +85,32 @@
 # define nxmq_pollnotify(msgq, eventset)
 #endif
 
-# define MQ_WNELIST(mq)               (&((mq)->waitfornotempty))
-# define MQ_WNFLIST(mq)               (&((mq)->waitfornotfull))
+# define MQ_WNELIST(cmn)              (&((cmn).waitfornotempty))
+# define MQ_WNFLIST(cmn)              (&((cmn).waitfornotfull))
 
 /****************************************************************************
  * Public Type Declarations
  ****************************************************************************/
 
+/* Common prologue of all message queue structures. */
+
+struct mqueue_cmn_s
+{
+  dq_queue_t waitfornotempty; /* Task list waiting for not empty */
+  dq_queue_t waitfornotfull;  /* Task list waiting for not full */
+  int16_t nwaitnotfull;       /* Number tasks waiting for not full */
+  int16_t nwaitnotempty;      /* Number tasks waiting for not empty */
+};
+
 /* This structure defines a message queue */
 
 struct mqueue_inode_s
 {
+  struct mqueue_cmn_s cmn;    /* Common prologue */
```

Comment:
Can we name it something like `struct mqueue_wait_node_s wnode;`? This is just a suggestion.

Reply:
Ok. If that is already used then let's keep it

Reply:
Done